libbeat/cfgfile/glob_manager.go

```diff
@@ -0,0 +1,168 @@
+package cfgfile
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// GlobManager allows to manage a directory of conf files. Using a glob pattern
+// to match them, this object will allow to switch their state between enabled
+// and disabled
+type GlobManager struct {
+	glob              string
+	enabledExtension  string
+	disabledExtension string
+	files             []*cfgfile
+}
+
+type cfgfile struct {
+	name    string
+	path    string
+	enabled bool
+}
+
+// NewGlobManager takes a glob and enabled/disabled extensions and returns a GlobManager object.
+// Parameters:
+//  - glob - matching conf files (ie: modules.d/*.yml)
+//  - enabledExtension - extension for enabled confs, must match the glob (ie: .yml)
+//  - disabledExtension - extension to append for disabled confs (ie: .disabled)
+func NewGlobManager(glob, enabledExtension, disabledExtension string) (*GlobManager, error) {
+	if !strings.HasSuffix(glob, enabledExtension) {
+		return nil, errors.New("Glob should have the enabledExtension as suffix")
+	}
+
+	g := &GlobManager{
+		glob:              glob,
+		enabledExtension:  enabledExtension,
+		disabledExtension: disabledExtension,
+	}
+	if err := g.load(); err != nil {
+		return nil, err
+	}
+	return g, nil
+}
+
+func (g *GlobManager) load() error {
+	g.files = []*cfgfile{}
```

Comment:
I would not initialize this to an empty slice. I've found it's best to not distinguish between empty and nil slices. https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices